### ISSUE TITLE
Coding difficulty

### ISSUE THREAD

**Issue by Trooksy** [nus-cs2030/2122-s1#322]
What is the best thing you did to improve your coding skills and made you understand the concepts?

**Comment by BrendanCheong** [comment]
Practice

**Comment by Irminrics** [comment]
I have found that practicing helps me to understand the concepts taught a lot better. Referring to videos and tutorials online on topics that are unclear has also helped me to gain a better perspective on these topics.

**Comment by ryan195** [comment]
I might have skipped a few lectures but I'd always watch the recordings of the lectures, and I find that by doing so, I can go through the intuition behind what Prof is doing in a slower pace, and google stuff I'm foreign with (Like which library has what, and what the expected output type of the method Prof is going through is). I also personally found the functors and monads thing interesting even though it's not tested, really helps to put things in perspective.  

Also personally, I found the project really fun, and it was the catalyst that got me to put in the required time needed to pass the module (I started coding the project only 2 weeks ago though so its probably too late by now to try and catch up with all the concepts). If I had better time management, I would have chosen to start on the project earlier to pique my interest in the mod earlier.

**Comment by lulucopter** [comment]
Practice will really get you into knowing the correct syntax and functionalities of the different API functions

**Comment by eugenelmj** [comment]
Adding on to what has been said above about watching lectures, personally I think that on top of that, watching the recitations really helped a lot. In a way, you get to see actual examples of codes, and how Prof handles them concretely (and details the minute differences in codes that may just make or break your code with that very one difference), while explaining the broad concepts that are pertinent to each lecture that he has gone through. As such, if time permits, I would recommend spending time to watch the recitation videos (since more elaboration goes on rather than simply referring to the recitation guides).

(Also since Finals would involve writing snippets of codes, I would think that the recitations serve as good guides on how to write the relevant codes to answer specific qns!)

**Comment by DarrenCsAcc** [comment]
Practice practice and do questions on kattis

**Comment by yuehernkang** [comment]
just keep doing and doing

**Comment by va5hi** [comment]
yeah tbh after watching lectures i'm pretty lost but after doing labs and practicing a bunch of times the idea gets clearer. I would recommend watching the recitations for concepts you are a little lost in it really helps in gaining a clearer understanding on the topics in my opinion.

**Comment by Lan-Jingbo** [comment]
Practice makes perfect lah

**Comment by bhone99** [comment]
youtube

**Comment by pikawrex** [comment]
googling to clarify on concepts helps me a lot!

**Comment by josiahoo** [comment]
I know that we are already provided with slides, but I found that making my own notes / mindmaps helped me to grasp and internalise the concepts better. Practice helped too!!

**Comment by urcs2030** [comment]
mindmaps are a great way to go about things as well
